weather.py

```diff
@@ -0,0 +1,50 @@
+import os
+import datetime
+import requests
+import sqlite3 as lite
+
+# Connect to database
+con = lite.connect('weather.db')
+cur = con.cursor()
+
+# Cities dictionary - key names here are the same as the city column names in the database
+cities = { "new_york": '40.663619,-73.938589',
+            "boston": '42.331960,-71.020173',
+            "nashville": '36.171800,-86.785002',
+            "philadelphia": '40.009376,-75.133346',
+            "washington_dc": '38.904103,-77.017229'
+        }
+
+# Build query - https://api.forecast.io/forecast/APIKEY/LATITUDE,LONGITUDE,TIME
+def get_forecast_data(city, request_time):
+	base = "https://api.forecast.io/forecast/"
+	api_key = os.environ["FORECAST_KEY"] + "/"
+	query = cities[city] + "," + str(request_time)
+	return base + api_key + query
+
+# Get time for 30 days ago
+thirty_days_ago = datetime.datetime.now() - datetime.timedelta(days=30)
+
+# Repeat for 30 days (2 requests per day, one for each city)
+for i in range(30):
+	# Set request time based on i (incrementing 1 day each time)
+	request_time = thirty_days_ago + datetime.timedelta(days=i)
+
+	# Convert to unix time
+	request_time = request_time.strftime("%s")
+
+	# Add request time to table
+	with con:
+		cur.execute('INSERT INTO max_temperature (request_time) VALUES (?)', (request_time,))
+
+	# Get max temperature for each city in the cities dictionary
+	for city in cities.keys():
+		# Send request using URL built in get_forecast_data
+		r = requests.get(get_forecast_data(city, request_time))
+
+		# Pull out the maximum temperature for that day
+		max_temp = r.json()['daily']['data'][0]['temperatureMax']
+
+		# Add max_temp to database
+		with con:
```

Comment:
[context managers](http://effbot.org/zone/python-with-statement.htm) (the `with con: ` part) are nice when you're doing the thing within once, and only temporarily. they are mostly used to make your code more readable and idiomatic. because you make a database connection several times in this file, i'd prefer to just set the `con` variable outright (and then `.close()` it (I believe) when you're finished. not a blocker. just my instinct.